core/controllers/ingest.py

```diff
@@ -83,6 +85,18 @@ def ingest(body: dict, excel_file: FileStorage) -> tuple[dict, int]:
 
     except ValidationError as validation_error:
         return process_validation_failures(validation_error.validation_failures, ingest_dependencies.messenger)
+    except Exception as e:
+        failure_uuid = save_failed_submission(g.excel_file)
+        current_app.logger.error(
+            f"Uncaught ingest exception: {type(e).__name__}: {str(e)}, - failure_id={str(failure_uuid)}", exc_info=True
+        )
+        return {
+            "detail": f"Uncaught ingest exception: {type(e).__name__}: {str(e)}",
+            "id": failure_uuid,
+            "status": 500,
+            "title": "Internal Server Error",
+            "internal_errors": [],
+        }, 500
```

Comment:
Just a thought - this could be encapsulated in a function like `process_uncaught_exception` to keep this top level function easier to skim over

Reply:
also it's best practice to use a more descriptive variable name than `e`

Reply:
Makes a lot of sense, thanks Cyrus! Just making these changes and will push an update shortly.